src/app.component.tsx

```diff
@@ -8,13 +8,23 @@ import { RouteSelectDialog } from './components/route-select.dialog'
 import { useDocumentTitle } from './shared/document-title.hook'
 
 export function AppComponent() {
-  const { t } = useTranslation()
+  const { t, i18n } = useTranslation()
   useDocumentTitle(t('label.title'))
   const [isOpenRouteSelect, setIsOpenRouteSelect] = useState(false)
   // Maybe keep it in local storage?
   const [showUserLocation, setShowUserLocation] = useState(false)
   const [selectedRoutes, setSelectedRoutes] = useState<Set<string>>(new Set())
 
+  useEffect(() => {
+    const language =
+      localStorage.getItem('language') || navigator.language.split('-')[0]
+    const existingLanguage = i18n.hasResourceBundle(language, 'translation')
+      ? language
+      : 'en'
+    localStorage.setItem('language', existingLanguage)
+    i18n.changeLanguage(existingLanguage)
+  }, [])
```

Comment:
One more suggestion and this is done I think 🙂 
```suggestion
  }, [i18n])
```